app/layout.py

```diff
@@ -0,0 +1,82 @@
+"""Module for generating figure grid layout."""
+from dash import html  # type: ignore
+
+
+class GridBuilder:
+    """_summary_."""
+
+    def __init__(
+        self,
+        rows: int,
+        cols: int,
+        row_heights: list[str] = None,  # type: ignore
+        col_widths: list[str] = None,  # type:ignore
+    ):
+        """_summary_.
+
+        Args:
+            rows (int): _description_
+            cols (int): _description_
+            row_heights (list[str], optional): _description_. Defaults to None.
+            col_widths (list[str], optional): _description_. Defaults to None.
+        """
+        self.rows = rows
+        self.cols = cols
+        self.row_heights = row_heights
+        self.col_widths = col_widths
+        self.row_heights = row_heights
+        self.col_widths = col_widths
+        if not self.row_heights:
+            self.row_heights = [f"{100 / rows}vh"] * rows
+        if not self.col_widths:
+            self.col_widths = [f"{100 / cols}%" for _ in range(cols)]
+        self.grid_elements = [[None] * cols for _ in range(rows)]
+
+    def add_element(self, element: html.Div, row: int, col: int) -> None:
+        """_summary_.
+
+        Args:
+            element (html.Div): _description_
+            row (int): _description_
+            col (int): _description_
+
+        Raises:
+            ValueError: _description_
+        """
+        if 0 <= row < self.rows and 0 <= col < self.cols:
+            self.grid_elements[row][col] = element
+        else:
+            raise ValueError("Invalid row or column index.")
```

Comment:
I'd swap these around so you don't need the `else`. It's just a pattern that helps reduce nesting in more complex functions and is a good habit to get into
```suggestion
        if 0 > row >= self.rows and 0 > col >= self.cols:
            raise ValueError("Invalid row or column index.")
        self.grid_elements[row][col] = element
```